table/tables/partition.go

```diff
@@ -117,6 +117,48 @@ type PartitionExpr struct {
 	Expr expression.Expression
 	// Used in the range pruning process.
 	*ForRangePruning
+	// Used in the range column pruning process.
+	*ForRangeColumnPruning
+}
+
+// ForRangeColumnPruning is used for range partition pruning.
+type ForRangeColumnPruning struct {
+	LessThan []expression.Expression
+	Column   ast.ExprNode
+	MaxValue bool
+}
+
+func dataForRangeColumnPruning(ctx sessionctx.Context, pi *model.PartitionInfo, schema *expression.Schema, names []*types.FieldName, p *parser.Parser) (*ForRangeColumnPruning, error) {
+	col, err := parseExpr(p, pi.Columns[0].L)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var res ForRangeColumnPruning
+	res.Column = col
+	res.LessThan = make([]expression.Expression, len(pi.Definitions))
+	for i := 0; i < len(pi.Definitions); i++ {
+		if strings.EqualFold(pi.Definitions[i].LessThan[0], "MAXVALUE") {
+			// Use a bool flag instead of math.MaxInt64 to avoid the corner cases.
+			res.MaxValue = true
+		} else {
+			tmp, err := parseSimpleExprWithNames(p, ctx, pi.Definitions[i].LessThan[0], schema, names)
+			if err != nil {
+				return nil, err
+			}
+			res.LessThan[i] = tmp
+		}
+	}
+	return &res, nil
+}
+
+// parseSimpleExprWithNames parses simple expression string to Expression.
+// The expression string must only reference the column in the given NameSlice.
+func parseSimpleExprWithNames(p *parser.Parser, ctx sessionctx.Context, exprStr string, schema *expression.Schema, names types.NameSlice) (expression.Expression, error) {
```

Comment:
Why do we use this function instead of the `ParseSimpleExprsWithNames` in simple_rewriter.go?

Reply:
Because `ParseSimpleExprsWithName` create a new parser everytime and causes performance problem for our user.

Reply:
Can we modify the old function instead of create a new one?

Reply:
It's shared by some other code and involve more changes.
Change a exposed package API is not a good choice here.